### ISSUE TITLE
Geth 1.6.5 hangs throwing “unexpected fault address” fatal error

### ISSUE THREAD

**Issue by computerphysicslab** [ethereum/go-ethereum#14629]
#### System information

Geth version: 1.6.5
Geth
Version: 1.4.10-stable-5f55d95a
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.6.2
OS: linux
GOPATH=
GOROOT=/opt/google/go

OS & Version: Linux Ubuntu 14.04.4 LTS (GNU/Linux 3.13.0-77-generic x86_64)
Commit hash : (if `develop`)

#### Expected behaviour

The geth node should be up and running for days, weeks or months.

#### Actual behaviour

The geth node throws a fatal error and stops from time to time (around 3 or 4 days).

#### Steps to reproduce the behaviour

This is the command to launch the geth node:

nohup ./geth --testnet --rpc --rpcapi "db,eth,net,web3,personal" --rpcport "8545" --mine --etherbase "0x8c60d40a2e848251d139fc2b0b6b770bb3351ffd" --gasprice "1000000000" --bootnodes "enode://20c9ad97c081d63397d7b685a412227a40e23c8bdc6688c6f37e97cfbc22d2b4d1db1510d8f61e6a8866ad7f0e17c02b14182d37ea7c3c8b9c2683aeb6b733a1@52.169.14.227:30303,enode://6ce05930c72abc632c58e2e4324f7c7ea478cec0ed4fa2528982cf34483094e9cbc9216e7aa349691242576d552a2a56aaeae426c5303ded677ce455ba1acd9d@13.84.180.240:30303" &

#### Backtrace

````
unexpected fault address 0x7fcb56135308
fatal error: fault
[signal SIGSEGV: segmentation violation code=0x1 addr=0x7fcb56135308 pc=0x461137]

goroutine 2885500 [running]:
runtime.throw(0xea5c2e, 0x5)
	/home/travis/.gimme/versions/go1.8.3.linux.amd64/src/runtime/panic.go:596 +0x95 fp=0xc4217bf9f0 sp=0xc4217bf9d0
runtime.sigpanic()
	/home/travis/.gimme/versions/go1.8.3.linux.amd64/src/runtime/signal_unix.go:297 +0x28c fp=0xc4217bfa40 sp=0xc4217bf9f0
runtime.memmove(0xc4209eb400, 0x7fcb56135308, 0x40)
	/home/travis/.gimme/versions/go1.8.3.linux.amd64/src/runtime/memmove_amd64.s:184 +0x657 fp=0xc4217bfa48 sp=0xc4217bfa40
github.com/ethereum/go-ethereum/consensus/ethash.hashimoto(0xc4217bfd78, 0x20, 0x20, 0x3e49620c80aa9db9, 0x51ffff80, 0xc423de4d00, 0xc4217bfc48, 0x40b944, 0xd34da0, 0xc42f51a300, ...)
	/home/travis/gopath/src/github.com/ethereum/go-ethereum/consensus/ethash/algorithm.go:315 +0x36e fp=0xc4217bfbe0 sp=0xc4217bfa48
github.com/ethereum/go-ethereum/consensus/ethash.hashimotoFull(0x7fcb56000008, 0x147fffe0, 0x147fffe0, 0xc4217bfd78, 0x20, 0x20, 0x3e49620c80aa9db9, 0xc423de4ca0, 0x20, 0x20, ...)
	/home/travis/gopath/src/github.com/ethereum/go-ethereum/consensus/ethash/algorithm.go:358 +0xbe fp=0xc4217bfc58 sp=0xc4217bfbe0
github.com/ethereum/go-ethereum/consensus/ethash.(*Ethash).mine(0xc420d88a80, 0xc42bf7e360, 0x0, 0x3e49620c80a4adf4, 0xc42f51a300, 0xc42f51a360)
	/home/travis/gopath/src/github.com/ethereum/go-ethereum/consensus/ethash/sealer.go:130 +0x451 fp=0xc4217bff68 sp=0xc4217bfc58
github.com/ethereum/go-ethereum/consensus/ethash.(*Ethash).Seal.func1(0xc431029220, 0xc420d88a80, 0xc42bf7e360, 0xc42f51a300, 0xc42f51a360, 0x0, 0x3e49620c80a4adf4)
	/home/travis/gopath/src/github.com/ethereum/go-ethereum/consensus/ethash/sealer.go:72 +0x87 fp=0xc4217bffa8 sp=0xc4217bff68
runtime.goexit()

````

**Comment by Monkube** [comment]
I get the same error, when trying  to update Geth through my Mac Terminal.
I did not use my wallet for 8 months, but I want to access it again.

I run the command:
./geth --fast --cache 1024

For an hour or so, it's busy downloading the latest blockchain. But then the Unexpected Fault Address makes everything stop.

IF you found a solution, please let me know.

**Comment by maximilianwollnik** [comment]
It is not a solution but I had the same problem and I have deleted my complete chaindata folder and restarted the whole process. This workaround worked for. At least for me it looks like the db-files were corrupted.

**Comment by immesys** [comment]
I have gotten something similar to this with a fresh database

**Comment by linas** [comment]
This appears to be identical to #14552 and to #14800

**Comment by LinXueyuanStdio** [comment]
I get the same error when I run `miner.start(1)`.
emmm....

**Comment by HanaAirichi** [comment]
@LinXueyuanStdio  did you find the solution ?

**Comment by karalabe** [comment]
This issue should have been fixed somewhere in 1.8.1 or maybe a couple releases later. If you still see it, lease open a new issue.
